### PR TITLE
More tests for GitConfig

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: test/libgit2.jl [hunk 1]
```diff
@@ -57,6 +57,14 @@ end
     @test sig3.email == sig.email
 end
 
+@testset "Default config" begin
+    #for now just see that this works
+    cfg = LibGit2.GitConfig()
+    @test isa(cfg, LibGit2.GitConfig)
```
Comment (ararslan) [inline test/libgit2.jl]: I guess it doesn't hurt having this test, but couldn't one safely assume that the `LibGit2.GitConfig` constructor returns a `LibGit2.GitConfig` object? What else could you get?

Comment (kshyatt) [inline test/libgit2.jl]: Nothing, it was just a "did the thing silently do something insane" sort of test.

**Review by ararslan — APPROVED** [review]
64-bit AV failure is the usual, OS X failure is #17626.

File: test/libgit2.jl (hunk 1 above)
Comment (ararslan) [inline test/libgit2.jl]: I understand this well; silently doing something insane is my MO.

**Pull request merged**